tests/jerry/string-prototype-charcodeat.js

```diff
@@ -0,0 +1,106 @@
+// Copyright 2015 Samsung Electronics Co., Ltd.
+// Copyright 2015 University of Szeged.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// check properties
+assert(Object.getOwnPropertyDescriptor(String.prototype.charCodeAt, 'length').configurable === false);
+
+assert(Object.getOwnPropertyDescriptor(String.prototype.charCodeAt, 'length').enumerable === false);
+
+assert(Object.getOwnPropertyDescriptor(String.prototype.charCodeAt, 'length').writable === false);
+
+assert(String.prototype.charCodeAt.length === 1);
+
+// check empty string
+assert(isNaN(String.prototype.charCodeAt.call(new String())));
+
+// check NaN
+assert("hello world!".charCodeAt(0) === 104);
+
+assert("HELLO WORLD".charCodeAt(10) === 68);
+
+// check Object
+assert(String.prototype.charCodeAt.call({})  === 91);
+
+
+// simple checks
+assert("hello world!".charCodeAt(0) === 104);
+
+assert("hello world!".charCodeAt(1) === 101);
+
+// check +-Inf
+assert(isNaN("hello world!".charCodeAt(-Infinity)));
+
+assert(isNaN("hello world!".charCodeAt(Infinity)));
+
+assert("hello world!".charCodeAt(11) === 33);
+
+assert(isNaN("hello world!".charCodeAt(12)));
+
+// check negative
+assert(isNaN("hello world!".charCodeAt(-1)));
+
+assert(isNaN("hello world!".charCodeAt(-9999999)));
+
+assert("hello world!".charCodeAt(-0) === 104);
+
+// check undefined
+assert("hello world!".charCodeAt(undefined) === 104);
+
+// check booleans
+assert("hello world!".charCodeAt(true) === 101);
+
+assert("hello world!".charCodeAt(false) === 104);
+
+// check this is undefined
+try {
+  String.prototype.charCodeAt.call(undefined);
+  assert(false);
+} catch(e) {
+  assert(e instanceof TypeError);
+}
+
+// check this is null
+try {
+  String.prototype.charCodeAt.call(null);
+  assert(false);
+} catch(e) {
+  assert(e instanceof TypeError);
+}
+
+// check coercible - undefined
+try {
+  assert(true.charCodeAt());
```

Comment:
This testcase will not invoke the `charCodeAt` at all. I think you wanted to write: `String.prototype.charCodeAt.call(true)`